core/src/main/java/com/netflix/conductor/validations/WorkflowTaskTypeConstraint.java

```diff
@@ -69,6 +69,9 @@ public boolean isValid(WorkflowTask workflowTask, ConstraintValidatorContext con
                 case TaskType.TASK_TYPE_DECISION:
                     valid = isDecisionTaskValid(workflowTask, context);
                     break;
+                case TaskType.TASK_TYPE_SWITCH:
```

Comment:
I see that the validation for LAMBDA task is missing here, but this would be a good time to add validation for the INLINE task which replaces it.

Reply:
TBH, there is nothing much to validate for the `LAMBDA` and `INLINE` task. The expressions are part of the `InputParameters` and so, there are no special fields that are must and need to be validated for these type of tasks. That probably is the reason why `LAMBDA` did not have a validation.

Reply:
Validation of LAMBDA/INLINE task could check that the script expression is not an empty string. 
The validation for this is being performed within the `Lambda` task evaluation currently and fails the task but this should be an early signal and fail workflow registration instead.

Reply:
From what I understand, the `LAMBDA` and now `INLINE`, the expressions are passed as input parameters. The input parameters could be wired from the workflow inputs, or outputs from another previously executed task. This only means that the validation of the expressions happen at the time of execution and as part of `WorkflowTask` validation.

Reply:
@apanicker-nflx 
This check changes the behavior. This check acts as a gate and expects the expression to be specified as part of the WorkflowTask's `evaluationType` and `expression` and always wire them into input parameters. Without the check, the expression can be inlined in `InputParameters` or can be defined as part of `WorkflowTask` and wire them into `InputParameters`. 

I can make the change and document the expected behavior for `INLINE` task. Will leave the `LAMBDA` task's behavior as it is so as to maintain existing behavior.

Please let me know.

Reply:
Ah yes you are right, let's keep the behavior the same for now. We can make this change in the future, if the need arises.